canonical-web-and-design/maas-ui
# feat(ui): build read-only machine power config section

## Done

- Built read-only machine power config section
  - This replaces the old functionality of it always being a form, with form fields being enabled depending on whether the user was in edit mode.
  - Added new components for the read-only section, with common utils being extracted from PowerTypeFields
- Removed "disabled" functionality for PowerTypeFields, since now this can't be done anywhere in the UI

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Go to the config tab of a machine that is not in a pod
- Check that the power parameters render as a list and not as form fields
- Click "Edit" and check that the section is now a form
- Change the power type to LXD and choose to generate a certificate
- Submit the form and check that it closes after success, and the generated certificate is displayed with its metadata and a download button
- Go to the config tab of a LXD machine that is in a pod
- Check that there are only node-scoped fields displayed
- Click "Edit" and check that you do not get the option to change the certificate

## Fixes

Fixes canonical-web-and-design/app-tribe#745

## Screenshot
![Peek 2022-04-01 13-58](https://user-images.githubusercontent.com/25733845/161192412-46f2c010-4899-4b38-a513-aa8a1989e38e.gif)
